### PR TITLE
[JDK17&21 on Release Test] Convert many functional bundles test batches to use jdk21_zulu instead of jdk8

### PR DESCRIPTION
**Pull request by julschong**
@tinatian @vicnate5 
- first three commits are simple conversion from jdk8 to jdk21_zulu because they are only used by portal-release test batch
- 3rd commit about mariadb, `functional-bundle-tomcat-mariadb106-jdk8` is used somewhere else, so I added new test batch `functional-bundle-tomcat-mariadb106-jdk21_zulu`
- 4th and 5th are simple remove for jdk11. they are not used anywhere else.

I will ask our ci team for help to confirm correctness by generating job summary again.

### CODE REVIEW

**Comment by julschong** [comment]
Job Summary is here: https://test-5-1.liferay.com/userContent/jobs/test-portal-release/builds/886/job-summary/index.html
The only 2 test batches missing are the amazonlinux2. I corrected the mistake on [83d92e6](https://github.com/liferay-core-infra/liferay-portal/pull/7311/commits/83d92e6ed98dc031b4e61c366a4df14d365fcc66)

File: test.properties [hunk 1]
```diff
@@ -7705,7 +7705,7 @@ test.batch.run.property.query[functional-orcllinux9-tomcat90-mysql57-jdk8]=\
     test.batch.run.property.query[functional-bundle-alpine310-tomcat90-mysql57-jdk21_zulu][portal-release]=\
         (portal.smoke == true)
 
-    test.batch.run.property.query[functional-bundle-amazonlinux*-tomcat90-mysql57-jdk8][portal-release]=\
+    test.batch.run.property.query[functional-bundle-amazonlinux*-tomcat90-mysql57-jdk21_zulu][portal-release]=\
```
Comment (julschong) [inline test.properties:7706]: this is newly added

**Comment by julschong** [comment]
@tinatian @vicnate5 ready for review

**Comment by vicnate5** [comment]
I am worried the OS tests (e.g. amazonlinux) might fail if they don't have JDK 21 installed, but there isn't going to be quick fix for those. We should still merge and take a look at the first results. If the OS tests fails, we should just disable them and create a ticket for CI to install JDK 21 on those.

**Comment by julschong** [comment]
@vicnate5 I can try running it on a test PR

**Comment by julschong** [comment]
@tinatian @vicnate5
@brittneyq helped us test ran `functional-bundle-amazonlinux*` test batches: 
https://test-1-25.liferay.com/job/test-portal-release-downstream/34190/console
https://test-1-25.liferay.com/job/test-portal-release-downstream/34191/console

it looks like the images don't have java21_zulu and gave this error:
```
start-app-server-preserve-liferay-home:
     [echo] start.app.server.0.start.timestamp: 08-02-2024 17:30:05:872 PDT
     [echo] ## Using /opt/java/zulu21 as JAVA_HOME.
     [echo] Executing commands: 
     [echo] 
     [echo] docker exec test-1-25-79.lax.liferay.com.amazonlinux2 mkdir -p /tmp
     [echo] docker cp /opt/dev/projects/github/liferay-portal/script.sh test-1-25-79.lax.liferay.com.amazonlinux2:/tmp/script.sh
     [echo] docker exec test-1-25-79.lax.liferay.com.amazonlinux2 /bin/bash /tmp/script.sh
     [echo] docker exec --user root test-1-25-79.lax.liferay.com.amazonlinux2 rm -f /tmp/script.sh
     [echo] 
     [exec] Unrecognized option: -Xlog:gc:/tmp/tomcat-gc.log
     [exec] Error: Could not create the Java Virtual Machine.
     [exec] Error: A fatal exception has occurred. Program will exit.
   [delete] Deleting: /opt/dev/projects/github/liferay-portal/script.sh
[beanshell] Waiting for server startup message
     [echo] Waiting for http://test-1-25-79.lax.liferay.com.amazonlinux2:8080/web/guest to load
[beanshell] 2 hours 5 minutes until timeout.
[beanshell] The top level build is alive.
[beanshell] 1 hour 60 minutes until timeout.
[beanshell] 1 hour 55 minutes until timeout.
[beanshell] The top level build is alive.
```
I will resend this pull to leave these ones untouched and stay with jdk8.
We will need to request ci team to update the linux distro images to include jdk21_zulu

**Pull request closed without merging**